src/shaders/_prelude_terrain.vertex.glsl

```diff
@@ -22,9 +22,27 @@ uniform float u_exaggeration;
 uniform float u_meter_to_dem;
 uniform mat4 u_label_plane_matrix_inv;
 
+uniform vec3 u_tile_tl_up;
+uniform vec3 u_tile_tr_up;
+uniform vec3 u_tile_br_up;
+uniform vec3 u_tile_bl_up;
+
 uniform sampler2D u_depth;
 uniform vec2 u_depth_size_inv;
 
+vec3 elevationVector(vec2 pos) {
+    vec2 uv = pos / 8192.0;
+    float topLen = length(u_tile_tl_up);
+    float bottomLen = length(u_tile_bl_up);
```

Comment:
Could `topLen` and `bottomLen` be computed on the CPU side and passed as uniforms?

Reply:
This is a good idea. This function is calling for refactoring anyways. These up vectors should be unit vectors only and the magnitude should be passed via `u_pixels_per_meter` or something similar. Now the pixels-per-meter is encoded in to the vectors.

Reply:
Alright, I'll add it as a separate uniform. I'll try it first on the cleanup branch to make sure it's working correctly with the globe view as well.

Reply:
@mpulkki-mapbox @karimnaaji The changes can be reviewed in https://github.com/mapbox/mapbox-gl-js/pull/10967 
If it's approved I'll apply it on this branch too and run the test cases. Currently I think it's easier to review the changes there because you can see the results on the globe.